graph/src/blockchain/mod.rs

```diff
@@ -51,18 +50,18 @@ pub trait Block: Send + Sync {
 }
 
 pub trait Blockchain: Sized + Send + Sync + 'static {
-    type Block: Block;
+    type Block: Block + Clone;
```

Comment:
I've added a comment. In terms of performance I believe the impact is small because this is only cloned when reprocessing a block, but I agree that it's unfortunate to require this bound.